zucchini-ui-frontend/src/stepDefinitions/components/StepDefinitionsTable.jsx

```diff
@@ -0,0 +1,255 @@
+        const key = getClosestOffset(knownOffsets, argument.offset);
+        const value = argument.value;
+        if (groupedArgs[key] === undefined) {
+          groupedArgs[key] = [value];
+        } else {
+          // No value duplicates
+          if (!includes(groupedArgs[key], value)) {
+            groupedArgs[key].push(value);
+          }
+        }
+      })
+    });
+  }
+  return groupedArgs;
+}
+
+function getClosestOffset(knownOffsets, currentOffset) {
+  // Sometimes the offset returned by cucumber is slightly shifted due to non-capturing groups in regexes
+  // Ex: " this is a glue with non capturing groups?" -> The "s" in "groups" will shift ethe offset by one
+  if (includes(knownOffsets[currentOffset])) {
+    return currentOffset;
+  }
+  return knownOffsets.reduce((prev, curr) => Math.abs(curr - currentOffset) < Math.abs(prev - currentOffset) ? curr : prev);
+}
+
+function formatVariantsBlock(variants, variantStyle) {
+  return variants.map(variant => {
+    return <StepDefinitionsTableHighlightedTerm key={variant} text={variant} styleClass={variantStyle} />
+  });
+}
+
+function formatStepDefinition(stepDefinition) {
+
+  const keyword = stepDefinition.info.keyword;
+  const definition = truncate(stepDefinition.info.name, { 'length': 300 });
+  const args = stepDefinition.info.arguments;
+
+  const formattedText = [];
+  formattedText.push(<span key={Math.random()}><StepDefinitionsTableHighlightedTerm text={keyword} styleClass='step-definition-keyword' />&nbsp;</span>);
+
+  let startIdx = 0;
+  if (args !== undefined) {
+    args.forEach((arg, index) => {
+      formattedText.push(definition.substr(startIdx, arg.offset - startIdx));
+      formattedText.push(<StepDefinitionsTableHighlightedTerm key={index} text={arg.value} styleClass='step-definition-argument' />);
+      startIdx = arg.offset + arg.value.length;
+    });
+    formattedText.push(definition.substr(startIdx));
+  } else {
+    formattedText.push(definition)
+  }
+  return formattedText;
+}
+
+function formatStepDefinitionVariants(occurrences) {
+
+  const blockStyle = { float: 'left' };
+  const keywordStyle = 'step-definition-keyword step-definition-variant';
+  const argumentStyle = 'step-definition-argument step-definition-variant';
+  const definition = occurrences[0].info.name;
+  const keywords = [...new Set(occurrences.map(occurrence => { return occurrence.info.keyword }))];
+  const groupedArgs = groupDistinctArgumentsByOffset(occurrences);
+
+  const formattedText = [];
+  formattedText.push(<div style={blockStyle} key={Math.random()}>{formatVariantsBlock(keywords, keywordStyle)}</div>);
+
+  let startIdx = 0;
+  for (const offset in groupedArgs) {
+    const position = parseInt(offset);
+    if (groupedArgs.hasOwnProperty(position)) {
+      const variants = groupedArgs[position];
+      formattedText.push(<div style={blockStyle} key={Math.random()}>{definition.substr(startIdx, position - startIdx)}</div>);
```

Comment:
Never use a random for the key attribute in React. Use an index variable for all keys.

Reply:
OK